src/Manager.sol

```diff
@@ -486,6 +481,10 @@ abstract contract Manager is
             payable(msg.sender).transfer(excessValue);
         }
 
+        // query tokens decimals
+        (, bytes memory queriedDecimals) = _token.staticcall(abi.encodeWithSignature("decimals()"));
+        uint8 decimals = abi.decode(queriedDecimals, (uint8));
+
         // normalize amount decimals
         uint256 normalizedAmount = normalizeAmount(amount, decimals);
```

Comment:
In `transfer` we do this:
```
uint256 newAmount = deNormalizeAmount(normalizeAmount(amount, decimals), decimals);
```

This is to cut off any dust and we revert if the transfer amount includes any dust amount of tokens. But we denormalize back to the correct number of decimals so that we can call the `safeTransferFrom` and `burnFrom` methods properly.

Then in `_transfer` we perform the actual normalization to 8 decimals for the cross-chain transfer.

Reply:
Ah yeah makes sense, thanks for clarifying!

Reply:
I was just thinking that maybe to save gas we could do this normalisation in `transfer` instead (and store the normalised amount in the queue as well) to avoid having to fetch the decimals again?

Reply:
We can definitely do that to save gas. I had originally kept the normalization logic in `_transfer` to be sure that it happens every time and isolate the normalization assumption. If you think the trade-off to save gas is worth it I can certainly refactor.

Reply:
I've not checked all the implications of this so maybe there's another blocker but generally I'd vote for saving gas here. We'd have to make sure the getter of a queued transfer denormalises to avoid confusion

Reply:
Synced offline with @djb15 on this -- we verified the two cases:
1. Transfer isn't queued -- it would obviously be cheaper not to double call, but because the slot is hot the extra gas is pretty minimal.
2. Transfer is queued -- It's cheaper to call again rather than use extra storage for the token decimals

So we will keep this code as-is.